recline.js

```diff
@@ -0,0 +1,326 @@
+;(function ($) {
+
+    // Constants.
+    var MAX_LABEL_WIDTH = 77;
+    var LABEL_MARGIN = 5;
+    var FILE_SIZE_PREVIEW_LIMIT = 1048576;
+
+    // Undefined variables.
+    var dataset, views, datasetOptions, fileSize, fileType, router;
+    var dataExplorerSettings, state, $explorer, dataExplorer;
+
+    // Create drupal behavior
+    Drupal.behaviors.Recline = {
+      attach: function (context) {
+        $explorer = $('.data-explorer');
+        // Local scoped variables.
+        fileSize = Drupal.settings.recline.fileSize;
+        fileType = Drupal.settings.recline.fileType;
+        dataExplorerSettings = {
+          grid: Drupal.settings.recline.grid,
+          graph: Drupal.settings.recline.graph,
+          map: Drupal.settings.recline.map
+        };
+
+        // This is the very basic state collection.
+        state = recline.View.parseQueryString(decodeURIComponent(window.location.hash));
```

Comment:
I know that this isn't realistic for this version of recline but I think hash-routing should ultimately be handled by a router (standard solution)